shared_buffer_pool.go

```diff
@@ -114,46 +114,33 @@ type simpleSharedBufferChildPool interface {
 
 type bufferPool struct {
 	sync.Pool
+
+	defaultSize int
 }
 
 func (p *bufferPool) Get(size int) []byte {
 	bs := p.Pool.Get().(*[]byte)
 
+	// If default size is 0. It means this pool is fallback pool.
+	// Therefore, we need to make a new one if the requested size is larger than the buffer.
+	if p.defaultSize == 0 {
```

Comment:
It's okay if you want to keep this, but....it should be unnecessary, right?  `Get` will never be called with something greater than `defaultSize` except when `defaultSize==0`, right?

Reply:
It's not necessary, but I fixed it because it's a safer way. 55354169ca42b4f3abbbf78debe9a89c4cd0ac1f